packages/core/storage-js/src/lib/vectors/VectorBucketApi.ts

```diff
@@ -0,0 +1,211 @@
+import { DEFAULT_HEADERS } from './constants'
+import { isStorageVectorsError } from './errors'
+import { Fetch, post } from './fetch'
+import { resolveFetch } from './helpers'
+import {
+  ApiResponse,
+  VectorBucket,
+  ListVectorBucketsOptions,
+  ListVectorBucketsResponse,
+} from './types'
+
+/**
+ * API class for managing Vector Buckets
+ * Provides methods for creating, reading, listing, and deleting vector buckets
+ */
+export default class VectorBucketApi {
+  protected url: string
+  protected headers: { [key: string]: string }
+  protected fetch: Fetch
+  protected shouldThrowOnError = false
+
+  constructor(url: string, headers: { [key: string]: string } = {}, fetch?: Fetch) {
+    this.url = url.replace(/\/$/, '')
+    this.headers = { ...DEFAULT_HEADERS, ...headers }
+    this.fetch = resolveFetch(fetch)
+  }
+
+  /**
+   * Enable throwing errors instead of returning them in the response
+   * When enabled, failed operations will throw instead of returning { data: null, error }
+   *
+   * @returns This instance for method chaining
+   * @example
+   * ```typescript
+   * const client = new VectorBucketApi(url, headers)
+   * client.throwOnError()
+   * const { data } = await client.createVectorBucket('my-bucket') // throws on error
+   * ```
+   */
+  public throwOnError(): this {
+    this.shouldThrowOnError = true
+    return this
+  }
+
+  /**
+   * Creates a new vector bucket
+   * Vector buckets are containers for vector indexes and their data
+   *
+   * @param vectorBucketName - Unique name for the vector bucket
+   * @returns Promise with empty response on success or error
+   *
+   * @throws {StorageVectorsApiError} With code:
+   * - `S3VectorConflictException` if bucket already exists (HTTP 409)
+   * - `S3VectorMaxBucketsExceeded` if quota exceeded (HTTP 400)
+   * - `InternalError` for server errors (HTTP 500)
+   *
+   * @example
+   * ```typescript
+   * const { data, error } = await client.createVectorBucket('embeddings-prod')
+   * if (error) {
+   *   console.error('Failed to create bucket:', error.message)
+   * }
+   * ```
+   */
+  async createVectorBucket(vectorBucketName: string): Promise<ApiResponse<undefined>> {
```

Comment:
Can we remove `vector` from all method names to match the `StorageBucketApi` and `StorageAnalyticsApi`?